cmd/cone/terraform_gen.go

```diff
@@ -0,0 +1,189 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/conductorone/cone/pkg/client"
+	"github.com/conductorone/cone/pkg/resource"
+	"github.com/pterm/pterm"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"golang.org/x/exp/maps"
+	"golang.org/x/exp/slices"
+)
+
+var tempFile = "cone_temp.txt"
+var tempTfFile = "cone_temp.tf"
+
+var objects = []string{"policy", "app_entitlement"}
+
+func terraformGenCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "gen <object-name> <terraform-directory-path>",
+		Short: "Import all terraform resources for the specified object type",
+		RunE:  terraformGen,
+	}
+	addTfAppIdFlag(cmd)
+
+	return cmd
+}
+
+func writeToFile(filename, data string) error {
+	file, err := os.Create(filename)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	_, err = file.WriteString(data)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// TODO @anthony: this probably could be cleaned up if we made the template generic
+func getResourceMap(ctx context.Context, c client.C1Client, v *viper.Viper, object string) (map[string]resource.TemplateData, error) {
+	resources := make(map[string]resource.TemplateData)
+	if object == "app" || object == "*" {
+		apps, err := c.ListApps(ctx)
+		if err != nil {
+			return nil, err
+		}
+		for _, app := range apps {
+			tmplData := resource.AppTemplate{App: app}
+			resources[tmplData.GetOutputId()] = tmplData
+		}
+	}
+	if object == "policy" || object == "*" {
+		policies, err := c.ListPolicies(ctx)
+		if err != nil {
+			return nil, err
+		}
+		for _, policy := range policies {
+			tmplData := resource.PolicyTemplate{Policy: policy}
+			resources[tmplData.GetOutputId()] = tmplData
+		}
+	}
+
+	if object == "app_entitlement" || object == "*" {
+		appId := v.GetString(tfAppIdFlag)
+		if appId == "" {
+			return nil, errors.New("app-id flag is required for app_entitlement object")
+		}
+
+		entitlements, err := c.ListEntitlements(ctx, appId)
+		if err != nil {
+			return nil, err
+		}
+		for _, entitlement := range entitlements {
+			tmplData := resource.AppEntitlementTemplate{AppEntitlement: entitlement}
+			resources[tmplData.GetOutputId()] = tmplData
+		}
+	}
+	return resources, nil
+}
+
+func terraformGen(cmd *cobra.Command, args []string) error {
+	ctx, c, v, err := cmdContext(cmd)
+	if err != nil {
+		return err
+	}
+
+	if err := validateArgLenth(2, args, cmd); err != nil {
+		return err
+	}
+
+	object := args[0]
+	if !slices.Contains(objects, object) && object != "*" {
+		return fmt.Errorf("invalid object name, only support %v and * for all", objects)
+	}
+
+	terraformDir := args[1]
+	if _, err := os.Stat(terraformDir); err != nil {
+		return fmt.Errorf("terraform directory %s does not exist", terraformDir)
```

Comment:
Maybe create it at this point instead of erroring?

Reply:
We do not want to create it since it will not have the provider.tf and it will just fail later down the line